src/crypto.rs

```diff
@@ -627,7 +675,10 @@ impl PublicKey {
 impl PartialEq for PublicKey {
     fn eq(&self, other: &Self) -> bool {
         // key_id is derived from these fields, so we ignore it.
-        self.typ == other.typ && self.scheme == other.scheme && self.value == other.value
+        self.typ == other.typ
+            && self.scheme == other.scheme
+            && self.keyid_hash_algorithms == other.keyid_hash_algorithms
+            && self.value == other.value
```

Comment:
github isn't letting me comment on lines outside of this patch, but you need to also include `keyid_hash_algorithms` in the `impl hash::Hash for PublicKey` as well.